webapp/channels/src/packages/mattermost-redux/src/utils/post_utils.ts

```diff
@@ -140,16 +140,31 @@ function isJoinLeavePostForUsername(post: Post, currentUsername: string): boolea
         post.props.removedUsername === currentUsername;
 }
 
+export function isPostUploadingFile(post: Post): boolean {
+    return post.id === post.pending_post_id && post.file_client_ids !== undefined && post.file_client_ids.length !== 0;
+}
+
+export function isPostDangling(state: GlobalState, post: Post) {
```

Comment:
@hmhealey 

> If I'm reading this right, is a "dangling post" another name for being a reply to a pending post?

Yes, that's right.

>  If so, is that needed for this ticket, or can we remove that part of the PR?

I implemented this to fix the following issue.
https://github.com/mattermost/mattermost-webapp/pull/11370#issuecomment-1373916031

I think we need this feature since an alert appears when a user replies to a pending post.
Can we ignore such a case?

Reply:
Ah, I see. I was under the impression that we didn't allow you to open the RHS to a pending post, but maybe that's not actually the case since, unless there's a lot of lag, it's hard to open the RHS to reply to a post before it's created currently. It'll definitely be easier now if we hold the post while we wait for attachments to upload.

Perhaps, we can either prevent the user from opening a pending post in the RHS or just replying to it? It's not ideal UX, but I feel like it'd be much easier to prevent that case from being possible. That way, we can also focus this effort on just queuing posts with pending file attachments without worrying about solving the existing edge cases caused by being able to reply to pending posts.

I'll see if anyone else here has any ideas, but how would that sound to you?